src/doors/TrainDoor.cpp

```diff
@@ -0,0 +1,502 @@
+        ADD_PROPERTY(
+                PropertyInfo(Variant::FLOAT, "platform/max_shift"), "set_door_platform_max_shift",
+                "get_door_platform_max_shift");
+        ClassDB::bind_method(
+                D_METHOD("set_door_platform_speed", "platform_speed"), &TrainDoor::set_door_platform_speed);
+        ClassDB::bind_method(D_METHOD("get_door_platform_speed"), &TrainDoor::get_door_platform_speed);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::FLOAT, "platform/speed"), "set_door_platform_speed", "get_door_platform_speed");
+        ClassDB::bind_method(D_METHOD("set_mirror_max_shift", "mirror_max_shift"), &TrainDoor::set_mirror_max_shift);
+        ClassDB::bind_method(D_METHOD("get_mirror_max_shift"), &TrainDoor::get_mirror_max_shift);
+        ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "mirror/max_shift"), "set_mirror_max_shift", "get_mirror_max_shift");
+        ClassDB::bind_method(D_METHOD("set_mirror_vel_close", "mirror_vel_close"), &TrainDoor::set_mirror_vel_close);
+        ClassDB::bind_method(D_METHOD("get_mirror_vel_close"), &TrainDoor::get_mirror_vel_close);
+        ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "mirror/vel_close"), "set_mirror_vel_close", "get_mirror_vel_close");
+        ClassDB::bind_method(D_METHOD("set_door_needs_permit", "needs_permit"), &TrainDoor::set_door_needs_permit);
+        ClassDB::bind_method(D_METHOD("get_door_needs_permit"), &TrainDoor::get_door_needs_permit);
+        ADD_PROPERTY(PropertyInfo(Variant::BOOL, "needs_permit"), "set_door_needs_permit", "get_door_needs_permit");
+        ClassDB::bind_method(
+                D_METHOD("set_door_permit_light_blinking", "blinking_mode"),
+                &TrainDoor::set_door_permit_light_blinking);
+        ClassDB::bind_method(D_METHOD("get_door_permit_light_blinking"), &TrainDoor::get_door_permit_light_blinking);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::INT, "permit/light_blinking", PROPERTY_HINT_ENUM, "ContinuousLight,FlashingOnPermissionWithStep,FlashingOnPermission,FlashingAlways"), "set_door_permit_light_blinking",
+                "get_door_permit_light_blinking");
+
+        BIND_ENUM_CONSTANT(PERMIT_LIGHT_CONTINUOUS_LIGHT);
+        BIND_ENUM_CONSTANT(PERMIT_LIGHT_FLASHING_ON_PERMISSION_WITH_STEP);
+        BIND_ENUM_CONSTANT(PERMIT_LIGHT_FLASHING_ON_PERMISSION);
+        BIND_ENUM_CONSTANT(PERMIT_LIGHT_FLASHING_ALWAYS);
+
+        BIND_ENUM_CONSTANT(PLATFORM_ANIMATION_TYPE_SHIFT);
+        BIND_ENUM_CONSTANT(PLATFORM_ANIMATION_TYPE_ROTATE);
+
+        BIND_ENUM_CONSTANT(DOOR_SIDE_RIGHT)
+        BIND_ENUM_CONSTANT(DOOR_SIDE_LEFT)
+
+        BIND_ENUM_CONSTANT(NOTIFICATION_RANGE_LOCAL)
+        BIND_ENUM_CONSTANT(NOTIFICATION_RANGE_UNIT)
+        BIND_ENUM_CONSTANT(NOTIFICATION_RANGE_CONSIST)
+
+        BIND_ENUM_CONSTANT(DOOR_STATE_CLOSE)
+        BIND_ENUM_CONSTANT(DOOR_STATE_OPEN)
+    }
+    void TrainDoor::_do_fetch_state_from_mover(TMoverParameters *mover, Dictionary &state) {
```

Comment:
Hm... większość tych properties to nie jest stan, tylko konfiguracja. Konfiguracja jest dostępna w properties nodes. Np. `needs_permit` jest w property node, który jest aplikowany do konfiguracji movera w `_do_update_mover()`. A nawet jeśli miałoby to wracać w `state`, to nazwy są lekko niespójne (`needs_permit` vs `pemit_needed`).

Stanem bieżącym systemu drzwi są parametry "runtime", czyli np. czy są otwarte czy zamknięte. Albo jak bardzo otwarte/zamknięte (i to pewnie z podziałem na lewe i prawe).  Być może będzie to też `is_opening` oraz `is_closing`, czyli stan w którym drzwi aktualnie są otwierane/zamykane. Zaglądając w Mover widzę jeszcze (do sprawdzenia): door.position, door.step_position (step_unfolding/step_folding - może to coś ciekawego).

![image](https://github.com/user-attachments/assets/9feaa85e-4865-42a8-b092-3592e4265de3)

Reply:
Znalazłem jeszcze takie coś, ale nie wiem czy jest dostęp z zewnątrz: czy drzwi są aktualnie pod prądem:
![image](https://github.com/user-attachments/assets/b3703745-58a6-448d-bd14-14d972579dbb)